cpp/oneapi/dal/backend/primitives/sort/test/perf_sort_dpc.cpp

```diff
@@ -0,0 +1,295 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/dataframe.hpp"
+#include "oneapi/dal/test/engine/io.hpp"
+#include "oneapi/dal/test/engine/config.hpp"
+#include "oneapi/dal/test/engine/math.hpp"
+#include "oneapi/dal/backend/primitives/sort/sort.hpp"
+
+namespace oneapi::dal::backend::primitives::test {
+
+namespace te = dal::test::engine;
+namespace la = te::linalg;
+namespace de = dal::detail;
+
+template <typename T>
+struct type2str {
+    static const char* name;
+};
+
+#define INSTANTIATE_TYPE_MAP(T) \
+    template <>                 \
+    const char* type2str<T>::name = #T;
+
+INSTANTIATE_TYPE_MAP(float);
+INSTANTIATE_TYPE_MAP(double);
+INSTANTIATE_TYPE_MAP(std::uint8_t);
+INSTANTIATE_TYPE_MAP(std::uint16_t);
+INSTANTIATE_TYPE_MAP(std::uint32_t);
+INSTANTIATE_TYPE_MAP(std::uint64_t);
+INSTANTIATE_TYPE_MAP(std::int8_t);
+INSTANTIATE_TYPE_MAP(std::int16_t);
+INSTANTIATE_TYPE_MAP(std::int32_t);
+INSTANTIATE_TYPE_MAP(std::int64_t);
+
+template <typename TestType>
+class sort_with_indices_test : public te::policy_fixture {
+public:
+    using Float = std::tuple_element_t<0, TestType>;
+    using Index = std::tuple_element_t<1, TestType>;
+
+    auto allocate_arrays(Index elem_count) {
+        auto& q = this->get_queue();
+        auto val = ndarray<Float, 1>::empty(q, { elem_count }, sycl::usm::alloc::device);
+        auto ind = ndarray<Index, 1>::empty(q, { elem_count }, sycl::usm::alloc::device);
+
+        Index* ind_ptr = ind.get_mutable_data();
+        q.submit([&](sycl::handler& cgh) {
+             cgh.parallel_for(sycl::range<1>(elem_count), [=](sycl::item<1> item) {
+                 Index ind = item.get_id()[0];
+                 ind_ptr[ind] = ind;
+             });
+         }).wait_and_throw();
+
+        return std::make_tuple(val, ind);
+    }
+
+    auto allocate_vector_arrays(std::int64_t vector_count, Index elem_count) {
```

Comment:
`vector_array` is confusing, when I saw it first time, I had wrong impression.

```suggestion
    auto allocate_vector_of_arrays(std::int64_t vector_count, Index elem_count) {
```

Reply:
renamed